managed/src/main/java/com/yugabyte/yw/controllers/CertificateController.java

```diff
@@ -128,13 +129,34 @@ public Result getRootCert(UUID customerUUID, UUID rootCA) {
     }
   }
 
-  public Result list(UUID customerUUID) {
-    List<CertificateInfo> certs = CertificateInfo.getAll(customerUUID);
-    if (certs == null) {
-      return ApiResponse.error(BAD_REQUEST, "Invalid Customer UUID: " + customerUUID);
-    }
-    return ApiResponse.success(certs);
-  }
+	public Result list(UUID customerUUID) {
+		List<CertificateInfo> certs = CertificateInfo.getAll(customerUUID);
+		HashMap<String, Boolean> removable = new HashMap<String, Boolean>();
+		List<Universe> universes = Universe.getAll();
+		for (Universe universe : universes) {
+			try {
+				Universe universe_obj = Universe.get(universe.universeUUID);
+				UUID cert = universe_obj.getUniverseDetails().rootCA;
+				removable.put(cert.toString(), true);
+			} catch (NullPointerException a) {
```

Comment:
Do use exceptions for control flow.
`if (<null check>) continue;`
for where you want to check.
What can be a `null`?

Reply:
If the universe is not attached to any certificate ( without TLS enabled ) then we would get null for `universe_obj.getUniverseDetails()` It will not return Null instead raises Nullpointer exception.

Reply:
@mahendranbhat That is incorrect. `getUniverseDetails()` should not have anything to do with the certs. The rootCA might be null, the universe details will not. Also, I am not sure I like the naming. Doesn't the cert belonging to a universe make it non removable? And as @sb-yb said, you should use a hashSet instead of a map. If the cert is not in the set, we then mark it as removable in the corresponding object that you create for it.

Reply:
Either way even if this was indeed throwing NPE that is still not a good reason to use exception for control flow. 
Based on what @Arnav15 says you need to investigate why `getUniverseDetails` would throw in such case.  May be write a unittest to repro.
My original comment has a type it should say "DONT use exception for control flow"

Reply:
@Arnav15 @sb-yb my bad , I thought I am using` getCertificate()` method to get a certificate.  ` getCertificate()` would give a null pointer exception.

Reply:
also `getUniverseDetails().rootCA`  would be null.